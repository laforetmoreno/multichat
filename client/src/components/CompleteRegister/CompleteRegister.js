import React from 'react';
import { func, string, bool } from 'prop-types';

import Input from '../Input';
import Button from '../Button';
import Form from '../Form';

import styles from './CompleteRegister.module.scss';

const CompleteRegister = ({ onSubmit, title = 'Chat IO', btnLabel = 'Login', ...props }) => (
  <Form onSubmit={onSubmit} className={styles.wrapper}>
    <h3 className={styles.title}>{title}</h3>
    <div className={styles.inputContainer}>
      <Input name="email" className={styles.input} placeholder="Email" {...props} />
      <Input name="name" className={styles.input} placeholder="Nome" {...props} />
      <Input name="birthday" className={styles.input} placeholder="AniversÃ¡rio" {...props} />
    </div>
    <Button onClick={onSubmit} className={styles.btn}>
      {btnLabel}
    </Button>
  </Form>
);

CompleteRegister.propTypes = {
  onClick: func,
  title: string,
  completeSigin: bool,
  btnLabel: string,
};

export default CompleteRegister;
