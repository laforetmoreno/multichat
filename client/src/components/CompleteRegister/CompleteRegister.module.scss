@import '../../styles/variables';

.wrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-width: 300px;

  @media (max-width: 767px) {
    width: 100%;
  }

  .title {
    font-size: 26px;
    text-align: center;
  }

  .inputContainer {
    width: 100%;

    @media (min-width: 768px) {
      display: flex;
      justify-content: center;
    }
  }

  .input {
    margin: 0 10px;

    @media (max-width: 767px) {
      margin: 5px 0;
    }
  }

  .btn {
    margin-top: 15px;
    width: 100%;
  }
}
