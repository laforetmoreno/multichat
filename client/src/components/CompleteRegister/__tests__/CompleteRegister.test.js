import React from 'react';
import CompleteRegister from '../CompleteRegister';
import renderer from 'react-test-renderer';
import { unmountComponentAtNode } from "react-dom";

describe('<CompleteRegister />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CompleteRegister />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Unmount Component At Node ', () => {
    let container = null;
      beforeEach(() => {
        container = document.createElement(<CompleteRegister />);
        document.body.appendChild(container);
      });

      afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
      });
  });
});