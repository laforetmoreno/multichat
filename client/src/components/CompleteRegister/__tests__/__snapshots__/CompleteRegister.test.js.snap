// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompleteRegister /> renders correctly 1`] = `
<form
  className="wrapper"
>
  <h3
    className="title"
  >
    Chat IO
  </h3>
  <div
    className="inputContainer"
  >
    <input
      className="wrapper input"
      data-testid="input"
      name="email"
      placeholder="Email"
    />
    <input
      className="wrapper input"
      data-testid="input"
      name="name"
      placeholder="Nome"
    />
    <input
      className="wrapper input"
      data-testid="input"
      name="birthday"
      placeholder="AniversÃ¡rio"
    />
  </div>
  <button
    className="btn primary"
    data-testid="btn"
  >
    Login
  </button>
</form>
`;
